### PR TITLE
feat(overview): add soft navigation and selectable reference / 总览页支持软导航与可选硬件基准

### PR DESCRIPTION
**Pull request by edwingao28**
## Summary

- switch overview selectors through App Router transitions without full-page reloads
- support B200, MI355X, B300, GB200, and GB300 as the comparison reference
- preserve the selected reference across SLO, engine scope, history mode, and locale URLs
- update deltas, reference styling, legends, and accessible labels against the selected hardware

Unofficial-run overlay: N/A. `/overview` is server-rendered and does not consume unofficial overlays.

## Validation

- 3,638 workspace unit tests
- 195 Cypress component tests
- 587 Cypress integration tests
- Overview E2E: 22/22 in Chrome and Firefox
- typecheck, lint, format, production fixture build

## 中文说明

- Overview 筛选项改用 App Router 软导航，切换时不再整页刷新
- 支持选择 B200、MI355X、B300、GB200 或 GB300 作为对比基准
- 在 SLO、引擎范围、历史对比和中英文 URL 间保留所选基准
- 根据所选硬件动态更新成本差异、基准列样式、图例和无障碍文案

非官方运行覆盖层：不适用。`/overview` 由服务端渲染，不读取非官方运行覆盖层。

验证：3,638 个 workspace 单元测试、195 个 Cypress 组件测试、587 个 Cypress 集成测试，以及 Chrome/Firefox 各 22 个 Overview E2E 均通过；typecheck、lint、format 与 production fixture build 通过。

---

> [!NOTE]
> **Medium Risk**
> Touches core overview pricing/delta assembly and client navigation state; behavior is well covered by unit and E2E tests but URL merging and transition timing could still cause edge-case desync.
> 
> **Overview**
> The overview page now changes **tier, engine scope, comparison mode, and reference hardware** through **App Router soft navigation** (`router.push` with `scroll: false`) instead of full document reloads. Controls stay real links for shareability; ordinary clicks go through `OverviewNavLink` / `OverviewNavigationProvider`, which merge pending query params via `mergeOverviewControlHref` so rapid successive changes don’t lose state while transitions are in flight. The header **language toggle** listens for `CLIENT_SEARCH_CHANGE_EVENT` so `?tier=`, `?ref=`, etc. stay correct when only search params change.
> 
> **Hardware comparison** is no longer fixed to B200: a new **`?ref=`** param (default B200) lets users pick among supported platforms. Server pages resolve `ref` and pass it into `getOverviewPageData`; cost deltas use **`costVsReferencePct`**, and the matrix highlights the reference column, updates legends/ARIA (∞ baseline), and embeds an **`OverviewReferenceSelect`** in the active “vs …” tab. Canonical URLs omit default `ref=b200`; other controls preserve `ref` across tier, engine, history mode, and locale links.
> 
> Cypress coverage adds scenarios for preserved window state on soft nav, rapid control changes, reference selection end-to-end, and rack SKU labels (e.g. GB200 NVL72).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 46a81537f9cae3f4fba8337af7de8e7215a7d24e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/app/src/components/overview/overview-reference-select.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      value={value}
+      onOpenChange={(open) => {
+        if (open) options.forEach((option) => navigation.prefetch(option.href, ['ref']));
+      }}
+      onValueChange={(next: OverviewReferenceHardware) => {
+        const option = options.find((candidate) => candidate.value === next);
+        if (option === undefined || next === value) return;
+        track('overview_reference_changed', { from: value, to: next });
+        navigation.push(option.href, ['ref']);
+      }}
```
[the remark this answers is not in the document]
Comment (edwingao28) [reply packages/app/src/components/overview/overview-reference-select.tsx:43]: Fixed in 40be9753 (#704). `OverviewReferenceSelect` no longer takes the reference from the server payload — it reads `useOverviewReference()`, which derives the value from the pending href (`overview-reference-select.tsx:33`).

The premise was correct at the time: `ref` round-tripped through the server for a value the client can resolve on its own. #704 removed that round trip entirely — a reference change now issues zero requests and the trigger updates on the same render as the URL, so there is nothing left to snap back from. `ref` is also excluded from the data cache key, and the failure-rollback path preserves a reference chosen mid-flight.
